src/ttl_track.rs

```diff
@@ -0,0 +1,546 @@
+            let conn_record = TcpConnRecord {
+                key,
+                time: SystemTime::now(),
+                ttl: ttl as u16,
+            };
+            self.conn_track.insert(key, conn_record);
+            true
+        } else {
+            false
+        }
+    }
+
+    // Cleanup expired connection records
+    fn tcp_cleanup(&mut self) {
+        let now = SystemTime::now();
+        if now
+            .duration_since(self.last_cleanup)
+            .unwrap_or(Duration::new(0, 0))
+            .as_secs()
+            >= TCP_CLEANUP_INTERVAL_SEC
+        {
+            self.last_cleanup = now;
+            self.conn_track.retain(|_, conn_record| {
+                now.duration_since(conn_record.time)
+                    .unwrap_or(Duration::new(0, 0))
+                    .as_secs()
+                    < TCP_CLEANUP_INTERVAL_SEC
+            });
+        }
+    }
+
+    // Handle incoming connection
+    pub fn tcp_handle_incoming(
+        &mut self,
+        src_ip: [u32; 4],
+        dst_ip: [u32; 4],
+        src_port: u16,
+        dst_port: u16,
+        is_ipv6: u8,
+        ttl: u8,
+    ) -> bool {
+        self.tcp_cleanup();
+        self.add_tcp_conn_track(&src_ip, &dst_ip, src_port, dst_port, is_ipv6, ttl)
+    }
+
+    // Handle outgoing connection
+    pub fn tcp_handle_outgoing(
+        &mut self,
+        src_ip: &[u32; 4],
+        dst_ip: &[u32; 4],
+        src_port: u16,
+        dst_port: u16,
+        conn_info: &mut TcpConnTrackInfo,
+        is_ipv6: u8,
+    ) -> bool {
+        let mut key = [0u8; TCP_CONN_RECORD_KEY_LEN];
+        Self::construct_key(dst_ip, src_ip, dst_port, src_port, &mut key, is_ipv6);
+
+        // If record exists, deconstruct and remove record
+        if let Some(conn_record) = self.check_get_tcp_conn_track_key(&key) {
+            Self::deconstruct_key(&key, conn_record, conn_info);
+            self.conn_track.remove(&key);
+            true
+        } else {
+            false
+        }
+    }
+}
+
+// Calculate TTL for fake packet
+pub fn tcp_get_auto_ttl(ttl: u8, auto_ttl_1: u8, auto_ttl_2: u8, min_hops: u8, max_ttl: u8) -> u8 {
+    let n_hops = if ttl > 98 && ttl < 128 {
+        128 - ttl
+    } else if ttl > 34 && ttl < 64 {
+        64 - ttl
+    } else {
+        return 0;
+    };
+
+    if n_hops <= auto_ttl_1 || n_hops < min_hops {
+        return 0;
+    }
+
+    let mut ttl_of_fake_packet = n_hops - auto_ttl_2;
+    if ttl_of_fake_packet < auto_ttl_2 && n_hops <= 9 {
+        ttl_of_fake_packet = n_hops
+            - auto_ttl_1
+            - ((auto_ttl_2 - auto_ttl_1) as f32 * (n_hops as f32 / 10.0)).trunc() as u8;
+    }
+
+    if max_ttl != 0 && ttl_of_fake_packet > max_ttl {
+        ttl_of_fake_packet = max_ttl;
+    }
+
+    ttl_of_fake_packet
+}
+
+// Set HTTP fragment size option
+static mut HTTP_FRAGMENT_SIZE: Option<u32> = None;
+
+pub fn set_http_fragment_size_option(fragment_size: u32) {
+    unsafe {
+        match HTTP_FRAGMENT_SIZE {
```

Comment:
Why we use unsafe codes here? I think here should a function with mutable var passes in but not use an unsafe global defination.

Reply:
It here should use a global defination indeed, please use `LazyCell`(for Rust 1.80+) instead of unsafe mut.